File: python-leetcode/linked-list/203.移除链表元素.py
from typing import Optional

from type import ListNode


class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        # 定义一个虚拟头结点
        dummy = ListNode(next=head)
        curr = dummy

        while curr.next:
            if curr.next.val == val:
                curr.next = curr.next.next
            else:
                curr = curr.next
        return dummy.next

File: go-leetcode/linked-list/203.移除链表元素.go
package leetcode

func removeElements(head *ListNode, val int) *ListNode {
	// 定义一个虚拟头结点
	dummy := &ListNode{Next: head}
	curr := dummy

	for curr.Next != nil {
		if curr.Next.Val == val {
			curr.Next = curr.Next.Next
		} else {
			curr = curr.Next
		}
	}
	return dummy.Next
}
